components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/utils/MLUtils.java

```diff
@@ -515,4 +520,48 @@ public static String getErrorMsg(String customMessage, Exception ex) {
         return null;
     }
 
+    /**
+     * Utility method to get the link to model build result page
+     *
+     * @param context ML model configuration context
+     * @return link to model build result page
+     * @throws DatabaseHandlerException
+     */
+    public static String getLink(MLModelConfigurationContext context, String status) throws DatabaseHandlerException {
+
+        MLModelData mlModelData = context.getModel();
+        long modelId = mlModelData.getId();
+        String modelName = mlModelData.getName();
+        long analysisId = mlModelData.getAnalysisId();
+        int tenantId = mlModelData.getTenantId();
+        String userName = mlModelData.getUserName();
+
+        MLAnalysis analysis = null;
+        String analysisName = null;
+        MLProject mlProject = null;
+        String projectName = null;
+        long datasetId;
+        DatabaseService databaseService = MLCoreServiceValueHolder.getInstance().getDatabaseService();
+        try {
+            analysis = databaseService.getAnalysis(tenantId, userName, analysisId);
+            analysisName = analysis.getName();
+            long projectId = analysis.getProjectId();
+
+            mlProject = databaseService.getProject(tenantId, userName, projectId);
+            projectName = mlProject.getName();
+            datasetId = mlProject.getDatasetId();
+        } catch (DatabaseHandlerException e) {
```

Comment:
@nuwand Fixed in pull request https://github.com/wso2/carbon-ml/pull/51. Thanks for pointing this out!